L1Trigger/TrackFindingTMTT/interface/Stub.h

```diff
@@ -0,0 +1,401 @@
+    }
+
+  private:
+    const Settings* settings_;  // configuration parameters.
+
+    unsigned int index_in_vStubs_;  // location of this stub in InputData::vStubs
+
+    //--- Parameters passed along optical links from PP to MP (or equivalent ones if easier for analysis software to use).
+    // WARNING: If you add any variables in this section, take care to ensure that they are digitized correctly by Stub::digitize().
+    float phi_;  // stub coords, optionally after digitisation.
+    float r_;
+    float z_;
+    float bend_;                           // bend of stub.
+    float dphiOverBend_;                   // related to rho parameter.
+    float dphiOverBendCorrection_;         // Correction from tilt of module
+    float dphiOverBendCorrection_approx_;  // Correction from tilt of module
+    unsigned int min_qOverPt_bin_;         // Range in q/Pt bins in HT array compatible with stub bend.
+    unsigned int max_qOverPt_bin_;
+
+    //--- Info about the two clusters that make up the stub.
+    array<float, 2> localU_cluster_;
+    array<float, 2> localV_cluster_;
+
+    //--- Parameters common to all stubs in a given module.
+    DetId idDet_;
+    float rErr_;
+    float zErr_;
+    float moduleMinR_;
+    float moduleMaxR_;
+    float moduleMinPhi_;
+    float moduleMaxPhi_;
+    float moduleMinZ_;
+    float moduleMaxZ_;
+    float moduleTilt_;
+    float pitchOverSep_;
+    unsigned int iphi_;
+    float alpha_;
+    bool psModule_;
+    unsigned int layerId_;
+    unsigned int endcapRing_;
+    bool barrel_;
+    bool tiltedBarrel_;
+    float sigmaPerp_;
+    float sigmaPar_;
+    float stripPitch_;
+    float stripLength_;
+    unsigned int nStrips_;
+    float sensorWidth_;
+    bool outerModuleAtSmallerR_;
+    //--- Truth info about stub.
+    const TP* assocTP_;
+    set<const TP*> assocTPs_;
+    //--- Truth info about the two clusters that make up the stub
+    array<const TP*, 2> assocTPofCluster_;
+
+    // Would front-end electronics output this stub?
+    bool frontendPass_;
+    // Did stub fail window cuts assumed in DegradeBend.h?
+    bool stubFailedDegradeWindow_;
+    // Bend in front end chip (prior to degredation by loss of bits & digitization).
+    float bendInFrontend_;
+    // Used for stub bend resolution degrading.
+    unsigned int numMergedBend_;
+
+    DigitalStub digitalStub_;   // Class used to digitize stub if required.
+    bool digitizedForGPinput_;  // Has this stub been digitized for GP input?
+    bool digitizedForHTinput_;  // Has this stub been digitized for HT input?
+    string
+        digitizedForSForTFinput_;  // Has this stub been digitized for seed filter or track fitter input? If so, this was its name.
+    bool digitizedForDRinput_;  // Has this stub been digitized for seed filter input?
+    bool digitizeWarningsOn_;   // Enable warnings about accessing non-digitized quantities.
+
+    // Which tracker geometry is this?
+    static thread_local string trackerGeometryVersion_;
```

Comment:
Changed to "static std::atomic<unsigned int>"

Reply:
What is the benefit of it being `static` over being just regular member variable? Ok, it is technically thread-safe now, but it smells like a suboptimal code structure.

Taking a look how this is now set in
https://github.com/cms-sw/cmssw/blob/3e5dc9509520c67f0cad13cc069ab231c8353845/L1Trigger/TrackFindingTMTT/src/Stub.cc#L755-L786
this kind of code would be better done with either [edm::ESWatcher](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_each_time_a_Record), or be moved in a separate ESProducer. (granted, the practical effect may sound negligible given that the tracker geometry is unlikely to change within a job, but I nevertheless believe that using the framework mechanisms would lead to more future-proof code).

Reply:
Noted. With regards to this particular bit of code, variable trackerGeometryVersion_ has been deleted.

Reply:
Dependence on trackerGeometryVersion variable has been eliminated.